packages/core/test/actor.test.ts

```diff
@@ -114,7 +114,7 @@ describe('spawning machines', () => {
       init: {
         entry: [
           assign({
-            server: () => spawn(serverMachine)
+            server: () => spawn(serverMachine) as Actor
```

Comment:
this should be fixable on our side, right? we shouldn't require people using `as` assertion to use `spawn`

Reply:
The problem is that it doesn't think an `Interpreter` is an `Actor`, even though `Interpreter implements Actor`. Any thoughts?

Reply:
Hm, seems like this should be a classic Animal vs Dog class problem and that Dog should always be assignable to Animal - such assignment should just lose extra (Dog-specific) information. If this fails for this reason - then something would be quite wrong with how TS interprets this? I might miss something about this situation of course.

Maybe you could try out if you can specify return type to both Actor & Interpreter (instead of asserting the return expression value) to see if it makes TS any happier? Maybe that would tell us something more about how it treats this? 🤷‍♂Alternatively, you could also try to specify a generic argument for `assign`. This still wouldn't improve the current situation anyhow, just maybe would help us to understand this.

The separate question is if it's a good thing that `spawn` returns different types. Maybe the could be unified? It seems though that accessing interpreter-specific stuff is valuable 🤔 So not sure how this could be simplified right now - although we maybe should assess how this can be accessed and if it's a good thing to access a child's instance? This could make also a great docs section - I think people might be a little bit confused about this (including myself 😉 )

Reply:
Resolved by making `state?` optional in `Actor`